pkg/auth/audit/audit.go

```diff
@@ -208,3 +212,41 @@ func isExist(array []string, key string) bool {
 	}
 	return false
 }
+
+func (a *auditLog) concealSensitiveData(requestURI string, body []byte) []byte {
+	var m map[string]interface{}
+	if err := json.Unmarshal(body, &m); err != nil {
+		return body
+	}
+
+	if strings.HasSuffix(requestURI, "secrets") {
+		// Conceal values of secret data.
+		dataKey := "data"
+		data, _ := m[dataKey].(map[string]interface{})
+		if data == nil {
+			dataKey = "stringData"
+			data, _ = m[dataKey].(map[string]interface{})
+		}
+
+		for key := range data {
+			data[key] = redacted
+		}
+		if data != nil {
+			m[dataKey] = data
+		}
+	}
+
+	for key := range m {
+		if _, ok := m[key].(string); ok {
+			if a.keysToConcealRegex.MatchString(key) {
+				m[key] = redacted
+			}
+		}
+	}
+
+	newBody, err := json.Marshal(m)
```

Comment:
You could track whether anything has been concealed and if not just return the input body without marshaling.

Reply:
I made this change.